f8a_worker/workers/graph_importer_utils.py

```diff
@@ -0,0 +1,204 @@
+                    if 'analyses' in obj:
+                        obj.get('analyses', {}).update(ver_obj['analyses'])
+                    else:
+                        obj.update(ver_obj)
+
+                # Check Package related information and add it to package object
+                if len(contents.get('pkg_list_keys')) > 0:
+                    pkg_obj = _other_key_info(contents.get('pkg_list_keys'), "package")
+                    if 'analyses' in obj:
+                        obj.get('analyses', {}).update(pkg_obj['analyses'])
+                    else:
+                        obj.update(pkg_obj)
+
+                # Create Gremlin Query
+                str_gremlin = GraphPopulator.create_query_string(obj)
+
+                if str_gremlin:
+                    # Fire Gremlin HTTP query now
+                    logger.info("Ingestion initialized for EPV - " +
+                                obj.get('ecosystem') + ":" + obj.get('package') + ":" + obj.get('version'))
+                    epv.append(obj.get('ecosystem') + ":" + obj.get('package') + ":" + obj.get('version'))
+                    payload = {'gremlin': str_gremlin}
+                    response = requests.post(GREMLIN_SERVER_URL_REST, data=json.dumps(payload), timeout=30)
+                    resp = response.json()
+
+                    if 'status' in resp and resp['status']['code'] == 200:
+                        count_imported_EPVs += 1
+                        last_imported_EPV = obj.get('ecosystem') + ":" + obj.get('package') + ":" + obj.get('version')
+                    elif 'Exception-Class' in resp:
+                        report['message'] = "The import failed " + resp['message']
+                        report['status'] = 'Failure'
+
+            except Exception as e:
+                msg = _get_exception_msg("The import failed", e)
+                report['status'] = 'Failure'
+                report['message'] = msg
+                report['epv'] = epv_key
+
+    report['epv'] = epv_list
+    report['count_imported_EPVs'] = count_imported_EPVs
+    if count_imported_EPVs == 0 and report['status'] == 'Success':
+        report['message'] = 'Nothing to be synced to Graph!'
+    report['last_imported_EPV'] = last_imported_EPV
+
+    return report
+
+
+def import_epv_from_s3_http(list_epv, select_doc=None):
+
+    try:
+        global s3, s3_pkg
+        s3 = StoragePool.get_connected_storage('S3Data')
+        s3_pkg = StoragePool.get_connected_storage('S3PackageData')
+        # Collect relevant files from data-source and group them by package-version.
+        list_keys = []
+        for epv in list_epv:
+            dict_keys = {}
+            ver_list_keys = []
+            pkg_list_keys = []
+            pkg_key_prefix = ver_key_prefix = ""
+
+            if 'name' not in epv or 'ecosystem' not in epv:
+                continue
+            elif 'version' not in epv or epv.get('version') is None:
+                epv['version'] = ''
```

Comment:
This is more to do with graph syncs and anything to do with workers. This will help us to sync multiple EPVs (some for package, some for pacakge-version). This may be helpful to invoke standalone graphimporter workflow. I would rather keep it unless it really causes some issues that I might not able to follow.

Reply:
The same as above - I would rather keep code minimal and doing desired actions properly instead of introducing unnecessary bulking complexity.

Reply:
tend to disagree here as I fail to understand what part of it is becoming too complex.

Reply:
You have way too many conditions, that's why @fridex find code too complex. Sometimes it's really necessary but you can think of an easier way it's always a better option.

I would guess that cyclomatic complexity of this part is unnecessarily high, but I am not sure.